common/src/crypto/hpke/trait_impls.rs

```diff
@@ -90,6 +91,29 @@ impl<KT> PgHasArrayType for DecryptionKey<KT> {
     }
 }
 
+impl<KT> Type<Sqlite> for DecryptionKey<KT> {
+    fn type_info() -> SqliteTypeInfo {
+        <Vec<u8> as Type<Sqlite>>::type_info()
+    }
+}
+
+impl<KT> Encode<'_, Sqlite> for DecryptionKey<KT> {
+    fn encode_by_ref(
+        &self,
+        buf: &mut <Sqlite as Database>::ArgumentBuffer<'_>,
+    ) -> Result<IsNull, BoxDynError> {
+        let bytes = PhnxCodec::to_vec(self).map_err(BoxDynError::from)?;
+        <Vec<u8> as Encode<'_, Sqlite>>::encode_by_ref(&bytes, buf)
+    }
+}
+
+impl<'r, KT> Decode<'r, Sqlite> for DecryptionKey<KT> {
+    fn decode(value: <Sqlite as Database>::ValueRef<'r>) -> Result<Self, BoxDynError> {
+        let bytes = <Vec<u8> as Decode<'r, Sqlite>>::decode(value)?;
```

Comment:
Consider to use `let bytes: &[u8] = Decode::<Sqlite>::decode(value)?;`. This syntax is less scary (no lifetime), and also it does not copy the buffer, but deserializes directly from the slice.